packages/graphql-function-transformer/src/FunctionTransformer.ts

```diff
@@ -0,0 +1,184 @@
+    role = (name: string, region: string): any => {
+        return new IAM.Role({
+            RoleName: Fn.If(
+                ResourceConstants.CONDITIONS.HasEnvironmentParameter,
+                Fn.Join('-', [
+                    FunctionResourceIDs.FunctionIAMRoleID(name, region).slice(0, 28), // max of 64. 64-10-26-28 = 0
+                    Fn.GetAtt(ResourceConstants.RESOURCES.GraphQLAPILogicalID, 'ApiId'), // 26
+                    Fn.Ref(ResourceConstants.PARAMETERS.Env) // 10
+                ]),
+                Fn.Join('-', [
+                    FunctionResourceIDs.FunctionIAMRoleID(name, region).slice(0, 38), // max of 64. 64-26-38 = 0
+                    Fn.GetAtt(ResourceConstants.RESOURCES.GraphQLAPILogicalID, 'ApiId'), // 26
+                ])
+            ),
+            AssumeRolePolicyDocument: {
+                Version: "2012-10-17",
+                Statement: [{
+                    Effect: "Allow",
+                    Principal: {
+                        "Service": "appsync.amazonaws.com"
+                    },
+                    Action: "sts:AssumeRole"
+                }]
+            },
+            Policies: [{
+                PolicyName: "InvokeLambdaFunction",
+                PolicyDocument: {
+                    Version: "2012-10-17",
+                    Statement: [{
+                        Effect: "Allow",
+                        Action: [
+                            "lambda:InvokeFunction"
+                        ],
+                        Resource: [lambdaArnResource(name, region)]
+                    }]
+                }
+            }]
+        })
+    }
+
+    /**
+     * Creates a lambda data source that registers the lambda function and associated role.
+     */
+    datasource = (name: string, region: string): any => {
+        return new AppSync.DataSource({
+            ApiId: Fn.Ref(ResourceConstants.PARAMETERS.AppSyncApiId),
+            Name: FunctionResourceIDs.FunctionDataSourceID(name, region),
+            Type: "AWS_LAMBDA",
+            ServiceRoleArn: Fn.GetAtt(FunctionResourceIDs.FunctionIAMRoleID(name, region), "Arn"),
+            LambdaConfig: {
+                LambdaFunctionArn: lambdaArnResource(name, region)
+            }
+        }).dependsOn(FunctionResourceIDs.FunctionIAMRoleID(name, region))
+    }
+
+    /**
+     * Create a new pipeline function that calls out to the lambda function and returns the value.
+     */
+    function = (name: string, region: string): any => {
+        return new AppSync.FunctionConfiguration({
+            ApiId: Fn.Ref(ResourceConstants.PARAMETERS.AppSyncApiId),
+            Name: FunctionResourceIDs.FunctionAppSyncFunctionConfigurationID(name, region),
+            DataSourceName: FunctionResourceIDs.FunctionDataSourceID(name, region),
+            FunctionVersion: "2018-05-29",
+            RequestMappingTemplate: printBlock(`Invoke AWS Lambda data source: ${FunctionResourceIDs.FunctionDataSourceID(name, region)}`)(obj({
```

Comment:
TODO: The build process does not extract function templates and upload to S3. I will update this.

Reply:
This has since been updated.